# time-serie-RNAseq-analysis

**About the project**

During my internship in a team studying chronobiology, one of my tasks was to set up a workflow for analyzing RNA-seq data from time-series experiments. To achieve this, I relied on the documentation provided by the library preparation kit supplier, as well as publications and repositories available in the references.

**Workflow**


![Image](https://github.com/user-attachments/assets/7c41895f-0e3b-42f0-87f0-603642114655)

**References** 

Teboul, Michèle, et Franck Delaunay. « Les oscillations harmoniques des rythmes circadiens sortent de l’ombre ». médecine/sciences 39, nᵒ 6‑7 (juin 2023): 544‑50. https://doi.org/10.1051/medsci/2023079.

Wu, Gang, Ron C. Anafi, Michael E. Hughes, Karl Kornacker, et John B. Hogenesch. « MetaCycle: An Integrated R Package to Evaluate Periodicity in Large Scale Data ». Bioinformatics 32, nᵒ 21 (1 novembre 2016): 3351‑53. https://doi.org/10.1093/bioinformatics/btw405.

Weger, Benjamin D., Cédric Gobet, Fabrice P. A. David, Florian Atger, Eva Martin, Nicholas E. Phillips, Aline Charpagne, Meltem Weger, Felix Naef, et Frédéric Gachon. « Systematic Analysis of Differential Rhythmic Liver Gene Expression Mediated by the Circadian Clock and Feeding Rhythms ». Proceedings of the National Academy of Sciences 118, nᵒ 3 (19 janvier 2021): e2015803118. https://doi.org/10.1073/pnas.2015803118.

Sica, Valentina, Oleg Deryagin, Jacob G. Smith, et Pura Muñoz‐Canoves. « Circadian Transcriptome Processing and Analysis: A Workflow for Muscle Stem Cells ». FEBS Open Bio 13, nᵒ 7 (juillet 2023): 1228‑37. https://doi.org/10.1002/2211-5463.13629.

Pan, Yinghong, Heather Ballance, Huan Meng, Naomi Gonzalez, Sam-Moon Kim, Leymaan Abdurehman, Brian York, et al. « 12-h Clock Regulation of Genetic Information Flow by XBP1s ». Édité par Achim Kramer. PLOS Biology 18, nᵒ 1 (14 janvier 2020): e3000580. https://doi.org/10.1371/journal.pbio.3000580.

Thaben, Paul F., et Pål O. Westermark. « Detecting Rhythms in Time Series with RAIN ». Journal of Biological Rhythms 29, nᵒ 6 (décembre 2014): 391‑400. https://doi.org/10.1177/0748730414553029.

Hughes, Michael E., Katherine C. Abruzzi, Ravi Allada, Ron Anafi, Alaaddin Bulak Arpat, Gad Asher, Pierre Baldi, et al. « Guidelines for Genome-Scale Analysis of Biological Rhythms ». Journal of Biological Rhythms 32, nᵒ 5 (octobre 2017): 380‑93. https://doi.org/10.1177/0748730417728663.

Hughes, Michael E., John B. Hogenesch, et Karl Kornacker. « JTK_CYCLE: An Efficient Nonparametric Algorithm for Detecting Rhythmic Components in Genome-Scale Data Sets ». Journal of Biological Rhythms 25, nᵒ 5 (octobre 2010): 372‑80. https://doi.org/10.1177/0748730410379711.


https://github.com/naef-lab/dryR/tree/master
https://github.com/gangwug/MetaCycle
https://www.bioconductor.org/packages/release/bioc/html/rain.html
https://github.com/5c077/Circa-in-Silico/tree/master

